python-avd/pyavd/_eos_designs/structured_config/network_services/utils.py

```diff
@@ -402,3 +402,44 @@ def get_vlan_aware_bundle_rt(
             return f"{admin_subfield}:{rt_override}"
 
         return f"{admin_subfield}:{bundle_number}"
+
+    def get_vrf_router_id(
+        self: AvdStructuredConfigNetworkServices,
+        vrf: EosDesigns._DynamicKeys.DynamicNetworkServicesItem.NetworkServicesItem.VrfsItem,
+        router_id: str,
+        tenant_name: str,
+    ) -> str | None:
+        """
+        Determine the router ID for a given VRF based on its configuration.
+
+        Args:
+            vrf (VrfsItem): The VRF object containing OSPF/BGP and vtep_diagnostic details.
+            router_id (str): The router ID type specified for the VRF (e.g., "vtep_diagnostic", "main_router_id", "none", or an IPv4 address).
+            tenant_name (str): The name of the tenant to which the VRF belongs.
```

Comment:
I think we don't need the types are we are already getting them from the signature. @ClausHolbechArista to confirm this is the way we want to go.

```suggestion
            vrf: The VRF object containing OSPF/BGP and vtep_diagnostic details.
            router_id: The router ID type specified for the VRF (e.g., "vtep_diagnostic", "main_router_id", "none", or an IPv4 address).
            tenant_name: The name of the tenant to which the VRF belongs.
```